event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/AstyanaxEventWriterDAO.java

```diff
@@ -62,9 +62,17 @@ public void addAll(Multimap<String, ByteBuffer> eventsByChannel, @Nullable Event
                 Duration eventTtl = _channelConfiguration.getEventTtl(channel);
 
                 int remaining = events.size();
+                long remainingSizeInBytes = events.stream().mapToLong(event->event.array().length).sum();
+                long numOversizeEvents = events.stream().filter(event->event.array().length > Constants.MAX_SLAB_SIZE_IN_BYTES).count();
+
+                // If there are oversize events slab allocation will fail so throw an exception
+                if (numOversizeEvents > 0) {
+                    throw new RuntimeException("AstyanaxEventWriterDAO.addAll: events (documents) present that are larger than max size of " + Constants.MAX_SLAB_SIZE_IN_MEGABYTES + " MB");
```

Comment:
10 Mb is way to large for a single event. We should discuss offline what an appropriate size is. It is likely closer to 1 Mb, if not lower.

Reply:
Yes I agree this was just defensive coding, as any effort to enqueue a document that is larger than a slab will not be successful.